code/modules/virus2/isolator.dm

```diff
@@ -71,8 +71,9 @@
 						var/datum/data/record/R = null
 						if (ID in virusDB)
 							R = virusDB[ID]
-
-						var/mob/living/carbon/human/D = B.data["donor"]
+		
+						var/datum/weakref/A = B.data["donor"]
+						var/mob/living/carbon/human/D = A.resolve()

```

Comment:
This can, in theory, resolve to `null` as well, right? Would it be possible to easily guard against this?